File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/HelloWorldCommandModule.java
package ro.razvan.java.dagger.atm;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;
import dagger.multibindings.StringKey;

@Module
public interface HelloWorldCommandModule {

    @Binds
    @IntoMap
    @StringKey(HelloWorldCommand.COMMAND_KEY_HELLO_WORLD)
    Command bindHelloWorldCommand(HelloWorldCommand command);

}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/HelloWorldCommandModule.kt
package ro.razvan.kotlin.dagger.atm

import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap
import dagger.multibindings.StringKey

@Module
interface HelloWorldCommandModule {

    @Binds
    @IntoMap
    @StringKey(COMMAND_KEY_HELLO_WORLD)
    fun bindHelloWorldCommand(command: HelloWorldCommand): Command
}